### ISSUE TITLE
Bug in parsing URLs

### ISSUE THREAD

**Issue by dzrdmwjx** [jsdom/whatwg-url#50]
While playing with some jquery/ajax stuff in node.js, I think I’ve hit a bug in the whatwg-url module used by jsdom. It’s unable to parse certain valid URLs such as: "https://r3---sn-p5qlsnz6.googlevideo.com/" and throws "TypeError: Invalid URL". I think the problem is with the “r3---sn” part of the hostname. 

_jsdom version: 8.1.0
whatwg-url version: 1.0.1_

Following node.js code reproduces the issue:

```
var url = "https://r3---sn-p5qlsnz6.googlevideo.com/";
parseURL = require("whatwg-url").parseURL;
var result = parseURL(url);

```

Output:

```
../node_modules/whatwg-url/lib/url-state-machine.js:1151
    throw new TypeError("Invalid URL");
    ^
TypeError: Invalid URL
    at Object.module.exports.basicURLParse (../node_modules/whatwg-url/lib/url-state-machine.js:1151:11)
    at module.exports.parseURL (../node_modules/whatwg-url/lib/url-state-machine.js:1189:25)
    at Object.<anonymous> (test.js:5:14)

```

**Comment by domenic** [comment]
Thanks; this is waiting on a spec resolution at https://github.com/whatwg/url/issues/53#issuecomment-181528158.

**Comment by dzrdmwjx** [comment]
Wondering if there's an update on this issue...
If Google's Youtube CDN host names don't meet the spec, can you provide me a hint as to where exactly this check is implemented in the code? I'll try to tweak it in my installation to hopefully make it work with such URLs. Thanks!

**Comment by Sebmaster** [comment]
Seems like with the release of Unicode 9 this has been changed and these labels are now allowed. Will update soon-ish. If you can't wait, the validation is in the [tr46 package](https://github.com/Sebmaster/tr46.js/blob/master/index.js#L108).

**Comment by dzrdmwjx** [comment]
That was super-easy! Thanks a lot :)

**Comment by patrickhulce** [comment]
Any updates here? It's not clear to me how the line referenced in `tr46` solves the issue with the above YouTube URLs as it's [this check](https://github.com/Sebmaster/tr46.js/blob/master/index.js#L116) that seems to be the problem. The latest version of `whatwg-url` (4.7.0) still throws. There was a proposed `allowHyphens` option in the [spec discussion](https://docs.google.com/document/d/11PEww2N0PbXyPhbsCdW_PjD3BNgZMy5XHUv02SSXNqY/edit#) that would skip this check. Can that be made available?

**Comment by TimothyGu** [comment]
This should be fixed in https://github.com/jsdom/whatwg-url/pull/100.

**Comment by Sebmaster** [comment]
Yup. Should be fixed with 6.2.0+.
